src/VelocityUKF.cpp

```diff
@@ -3,7 +3,7 @@
 using namespace uwv_kalman_filters;
 
 const std::string VelocityUKF::acceleration_measurement = "acceleration";
```

Comment:
> From the point of the parent classes (AbstractFilter and UKF).

From what you're saying, they don't need to know what is what in the measurements, right ? It's all the same for it ?

If that's the case, you could have a:
 - integer in the measurement data structure that discriminates which measurement is which
 - restrict client classes (subclasses) to allocate these integers sequentially from zero
 - construct the base filter with the maximum number of measurements so that it can preallocate its own arrays at construction time.

The "client" class (i.e. the subclass) would then allocate these into an enum to have a nice mapping of integer-to-measurement. You could even declare the measurement names to the parent class.

Then all you need when you want to store per-measurement info (the way you do it in your class) is a vector allocated once and for all instead of having to get through maps all the time.

Why ?  Because all these strings allocate on the heap. Each time. For something that's essentially static. It restricts the use of these filters, and will make the whole filter chains flaky (at best) under high memory pressure.

Or am I missing something ?